crypto-test/src/test/scala/org/bitcoins/crypto/ECPrivateKeyTest.scala

```diff
@@ -117,4 +121,20 @@ class ECPrivateKeyTest extends BitcoinSUnitTest {
     }
   }
 
+  it must "correctly execute the ecdsa single signer adaptor signature protocol" in {
+    forAll(CryptoGenerators.privateKey,
+           CryptoGenerators.privateKey,
+           NumberGenerator.bytevector(32)) {
+      case (privKey, adaptorSecret, msg) =>
+        val adaptorSig = privKey.adaptorSign(adaptorSecret.publicKey, msg)
+        assert(
+          privKey.publicKey
+            .adaptorVerify(msg, adaptorSecret.publicKey, adaptorSig))
+        val sig = adaptorSecret.completeAdaptorSignature(adaptorSig)
+        val secret =
+          adaptorSecret.publicKey.extractAdaptorSecret(adaptorSig, sig)
+        assert(secret == adaptorSecret)
+        assert(privKey.publicKey.verify(msg, sig))
```

Comment:
Maybe add a check that `sig == privKey.sign(msg)`

Reply:
I don't think this will be true due to nonce selection being different